applications/mlflow/Taskfile.yml

```diff
@@ -47,9 +55,131 @@ tasks:
       - task --list
     silent: true
 
+  # Development environment tasks
+  dev:build-image:
+    desc: Build development Docker image
+    silent: true
+    cmds:
+      - docker build -t {{.DEV_IMAGE}} -f Dockerfile .
+
+  dev:start:
+    desc: Start development container in background
+    silent: true
+    cmds:
+      - echo "Starting development container..."
+      - |
+        # Check if container is running
+        if docker ps | grep -q "mlflow-dev-container"; then
+          echo "Container mlflow-dev-container is already running"
+        # Check if container exists but is not running
+        elif docker ps -a | grep -q "mlflow-dev-container"; then
+          echo "Container mlflow-dev-container exists but is not running. Starting it..."
+          docker start mlflow-dev-container
+        # Create and start new container
+        else
+          # Try ports 5000-5004, use the first available one
+          PORT_FOUND=false
+          for PORT in {5000..5004}; do
+            if ! lsof -i :$PORT > /dev/null; then
+              PORT_ARG="-p $PORT:$PORT"
+              echo "Using port $PORT for MLflow UI"
+              PORT_FOUND=true
+              break
+            fi
+          done
+
+          # If all standard ports are in use, use a random port
+          if [ "$PORT_FOUND" = "false" ]; then
+            PORT_ARG="-P"
+            echo "All standard ports are in use. Using a random port."
+          fi
+
+          # Start container with host networking for kubectl port-forward compatibility
+          CONTAINER_ID=$(docker run --name mlflow-dev-container --network host -d \
```

Comment:
Wouldn't it be better to `--rm` to allow cleaning after self?

Reply:
Can the command be abstracted, while docker is common so is podman it would be nice to allow the container system to be either one.

Reply:
> Wouldn't it be better to `--rm` to allow cleaning after self?

I ended up getting rid of `--rm` because i preferred to keep the container around and just `shell` back into it.

Reply:
> Can the command be abstracted, while docker is common so is podman it would be nice to allow the container system to be either one.

Looks like there is pretty good compatibility in podman with the `docker` CLI already. Would have to spend some additional time with podman myself to see if there's parity with what I have today or if I needed to actually abstract both `docker` and `podman` commands in this CLI .